helix-term/src/compositor.rs

```diff
@@ -27,6 +27,24 @@ pub struct Context<'a> {
     pub jobs: &'a mut Jobs,
 }
 
+impl<'a> Context<'a> {
+    /// Waits on all pending jobs, and then tries to flush all pending write
+    /// operations for all documents.
+    pub fn block_try_flush_writes(&mut self) -> anyhow::Result<()> {
+        tokio::task::block_in_place(|| {
+            helix_lsp::block_on(self.jobs.finish(Some(self.editor), None))
+        })?;
+
+        for doc in &mut self.editor.documents.values_mut() {
+            tokio::task::block_in_place(|| helix_lsp::block_on(doc.try_flush_saves()))
+                .map(|result| result.map(|_| ()))
+                .unwrap_or(Ok(()))?;
+        }
```

Comment:
Should we try to save all the files rather than bailing out on the first error? But I think this should be fine for now since it's rare to save multiple files and to fail.

Reply:
Possibly, although I don't know if it's worth the added complexity of the error handling at the call site. You'd have to do something like return a `Vec<Result<_>>`.

Reply:
I guess it is fine for now then.